rhythm.py

```diff
@@ -211,18 +217,17 @@ def wait_for_bell_time(self, current_time, bell, row_number, place, user_control
             # Slow the ticks slightly
             sleep(0.01)
 
-    def expect_bell(self, expected_bell, row_number, index, expected_stroke):
+    def expect_bell(self, expected_bell, row_number, place, expected_stroke):
```

Comment:
What do you think about a future change of this to `expect_row(self, row, row_number, expected_stroke)` instead of forcing the bot to enumerate the row

Reply:
I think this would be good, although `expect_bell` is only ever called from one function in Bot anyway, since I noticed that having it called from 3 different places was a refactoring liability.

On the other hand, this makes the change pretty straightforward - I could potentially do it in this PR?

Reply:
Sure, just note it's on the abstract base so you need to change that and `WaitForUserRhythm` as well

Reply:
Yep the linter check won't let me break that :)

Reply:
Having just started this, it's become clear that because of the abstract nature of the `Rhythm` class, this will become fairly ugly because every time expect_row is actually going to be used each concrete `Rhythm` class will have to enumerate the bells anyway and check that they're user defined (breaking the loose coupling between the `Rhythm` and the `Tower`).  So at this point, it makes more sense to just do this enumeration once in the `Bot` class - and all the calls to this are funnelled through `start_next_row` which makes sure the enumeration only appears once in the code.